db/patients.ts

```diff
@@ -17,17 +17,31 @@ const getById = async (patientId: string, psychologistId: string): Promise<Patie
   }
 };
 
-const getAll = async (psychologistId: string): Promise<(Patient & {appointmentsCount: string})[]> => {
+const startCurrentUnivYear = (): string => {
+  const currentMonth = new Date().getMonth();
+  const currentYear = new Date().getFullYear();
+  if (currentMonth < 6) {
+    return `${currentYear - 1}-09-01`;
+  }
+  return `${currentYear}-09-01`;
+};
+
+const getAll = async (psychologistId: string): Promise<(Patient & { appointmentsCount: string })[]> => {
   try {
     const patientArray = await db.select(`${patientsTable}.*`)
-        .from(patientsTable)
-        .joinRaw(`left join "${appointmentsTable}" on `
+      .from(patientsTable)
+      .joinRaw(`left join "${appointmentsTable}" on `
         + `"${patientsTable}"."id" = "${appointmentsTable}"."patientId" and "${appointmentsTable}"."deleted" = false`)
-        .where(`${patientsTable}.psychologistId`, psychologistId)
-        .andWhere(`${patientsTable}.deleted`, false)
-        .count(`${appointmentsTable}.*`, { as: 'appointmentsCount' })
-        .groupBy(`${patientsTable}.id`)
-        .orderByRaw(`LOWER("${patientsTable}"."lastName"), LOWER("${patientsTable}"."firstNames")`);
+      .where(`${patientsTable}.psychologistId`, psychologistId)
+      .andWhere(`${patientsTable}.deleted`, false)
+      .count(`${appointmentsTable}.*`, { as: 'appointmentsCount' })
+      .count(db.raw(
+        `DISTINCT CASE WHEN ${appointmentsTable}."appointmentDate" > '${startCurrentUnivYear()}' 
+        THEN ${appointmentsTable}.id 
+        END`,
+      ))
```

Comment:
I didn't find the way to add a "as" here, so this value name is `count` :/

Reply:
weird beacause you have a count just on top ? 
.count(`${appointmentsTable}.*`, { as: 'appointmentsCount' })

Reply:
Yes but the with the `raw` the syntaxt with `{as :` doesn't work :(

Reply:
nitpick: You can try having a `select(db.raw('count .... as ...'))` ?